src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -200,51 +200,19 @@ public virtual async Task<T> GetConfigurationAsync(CancellationToken cancel)
                 {
                     fetchMetadataFailure = ex;
 
-                    // In this case configuration was never obtained.
```

Comment:
As I understand, the logic in that catch is that


if we failed to retrieve the initial config, then we set _syncAfter to boostrapRefreshInterval.
then we throw an exception that we failed to retrieve config.
_bootstrapRefreshInterval is used because is much shorter (1 second) than our normal refresh (5 min).
If we retrieved the initial config but failed validating it
then we set _syncAfter to our normal refresh interval
and throw an exception IDX20806 that we failed to update the config.
I do agree that it seems like that code, and bootstrapRefreshInterval, is not needed because if we fail to retrieve the initial config, _currentConfiguration will still be null. So that the following call to GetConfigurationAsync will try to retrieve the config again, ignoring the _syncAfter value.

Reply:
@pmaytak if we failed to obtain configuration, we will still throw an exception with IDX20803, right at the end.
What i am thinking is that if we never received a valid configuration, why should we change any values?